src/utils/command_pipe.rs

```diff
@@ -52,107 +53,96 @@ impl CommandPipe {
         Ok(Self { pipe_file, rx })
     }
 
-    pub async fn read_command(&mut self) -> Option<ExternalCommand> {
+    pub async fn read_command(&mut self) -> Option<Command> {
         self.rx.recv().await
     }
 }
 
-async fn read_from_pipe(
-    pipe_file: &Path,
-    tx: &mpsc::UnboundedSender<ExternalCommand>,
-) -> Option<()> {
+async fn read_from_pipe(pipe_file: &Path, tx: &mpsc::UnboundedSender<Command>) -> Option<()> {
     let file = fs::File::open(pipe_file).await.ok()?;
     let mut lines = BufReader::new(file).lines();
 
     while let Some(line) = lines.next_line().await.ok()? {
-        let cmd = parse_command(&line).ok()?;
+        let cmd = match parse_command(&line) {
+            Ok(cmd) => cmd,
+            Err(err) => {
+                log::error!("An error occurred while parsing the command: {}", err);
+                return None;
+            }
+        };
         tx.send(cmd).ok()?;
     }
 
     Some(())
 }
 
-fn parse_command(s: &str) -> std::result::Result<ExternalCommand, ()> {
+fn parse_command(s: &str) -> Result<Command, Box<dyn std::error::Error>> {
     let head = *s.split(' ').collect::<Vec<&str>>().get(0).unwrap_or(&"");
     match head {
-        "UnloadTheme" => Ok(ExternalCommand::UnloadTheme),
-        "Reload" => Ok(ExternalCommand::Reload),
-        "ToggleFullScreen" => Ok(ExternalCommand::ToggleFullScreen),
-        "SwapScreens" => Ok(ExternalCommand::SwapScreens),
-        "MoveWindowToLastWorkspace" => Ok(ExternalCommand::MoveWindowToLastWorkspace),
-        "FloatingToTile" => Ok(ExternalCommand::FloatingToTile),
-        "MoveWindowUp" => Ok(ExternalCommand::MoveWindowUp),
-        "MoveWindowDown" => Ok(ExternalCommand::MoveWindowDown),
-        "FocusWindowUp" => Ok(ExternalCommand::FocusWindowUp),
-        "MoveWindowTop" => Ok(ExternalCommand::MoveWindowTop),
-        "FocusWindowDown" => Ok(ExternalCommand::FocusWindowDown),
-        "FocusNextTag" => Ok(ExternalCommand::FocusNextTag),
-        "FocusPreviousTag" => Ok(ExternalCommand::FocusPreviousTag),
-        "FocusWorkspaceNext" => Ok(ExternalCommand::FocusWorkspaceNext),
-        "FocusWorkspacePrevious" => Ok(ExternalCommand::FocusWorkspacePrevious),
-        "NextLayout" => Ok(ExternalCommand::NextLayout),
-        "PreviousLayout" => Ok(ExternalCommand::PreviousLayout),
-        "RotateTag" => Ok(ExternalCommand::RotateTag),
-        "CloseWindow" => Ok(ExternalCommand::CloseWindow),
-        // These require arguments and might be more finicky
-        "LoadTheme" => build_load_theme(s),
```

Comment:
If we lose `LoadTheme` and `UnloadTheme`, we cannot switch themes, without at least a `SoftReload`, right?
Personally I haven't used this possibility yet, but I believe it might actually be nice to keep this. Or am I missing the point where this is still possible, but the way its triggerd has changed?

Reply:
Yes it is still possible. When a command string is not recognized it is forwarded to Config::command_handler. This function works for both CommandPipe **AND** for keybind Command::Other. You will see that the code to do the theme load/unload is in `src/bin/common/config.rs`.

My idea is that, if I have a tiling manager library, there is no concept of "theme". It's a great concept for leftwm to abstract to the user that kind of configuration but it is an unneeded abstraction for the lib itself.